frontend/src/app/api/addToSheet/route.ts

```diff
@@ -0,0 +1,35 @@
+import { NextResponse } from 'next/server';
+import { google } from 'googleapis';
+
+export async function POST(request: Request) {
+  try {
+    const body = await request.json();
+    const { name, email, role } = body;
+
+    const auth = new google.auth.GoogleAuth({
+      credentials: {
+        client_email: process.env.GOOGLE_CLIENT_EMAIL,
+        private_key: process.env.GOOGLE_PRIVATE_KEY?.replace(/\\n/g, '\n'),
+      },
+      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+    });
+
+    const sheets = google.sheets({ version: 'v4', auth });
+
+    const spreadsheetId = process.env.SPREADSHEET_ID;
+
+    const response = await sheets.spreadsheets.values.append({
+      spreadsheetId,
+      range: 'Sheet1!A:C',
+      valueInputOption: 'USER_ENTERED',
+      requestBody: {
+        values: [[name, email, role]],
```

Comment:
add a created at date value so we can track when someone joined the waitlist.

Reply:
Done, I chose a date format that is recognizable by Google sheets, but it can be customized in the template